unity-client/Assets/Plugins/CustomRequestAnimationFrame/CustomRequestAnimationFrame.jspre

```diff
@@ -0,0 +1,69 @@
+
+(function () {
+  if (window.requestAnimationFrame && document) {
+    window.backgroundFPS = 30;
+    window.__requestAnimationFrame = window.requestAnimationFrame;
+    window.__CURRENT_RAFS = {};
+
+    window.unitySkippedLastRAF = true;
+    window.cachedUnityRAFCallback;
+
+    function backgroundRAF(rafCallback) {
+      setTimeout(function(){
+        rafCallback(performance.now())
+      }, 1000 / window.backgroundFPS);
+    }
+
+    function nonUnityRAFCallback(stamp) {
+      window.requestAnimationFrame(nonUnityRAFCallback); // ends up calling the normal RAF again
+    }
+
+    window.requestAnimationFrame = function (unityRAFCallback) { // Unity's rafCallback calls RAF inside
+      if (document.hidden) {
+        backgroundRAF(unityRAFCallback);
+      } else {
+        var rafId = __requestAnimationFrame(function(stamp) {
```

Comment:
Please take into account that rAF is not only used by unity, it is a globally available function and if some other part of the code calls it, the flip-flop logic implemented here would be broken for both codes.